### PR TITLE
feat: Dune Analytics Data Provider

### PR DESCRIPTION
**Pull request by sampolgar**
New PR including Dune Analytics and alphabet changes to GitHub workflow

### CODE REVIEW

File: group-generators/helpers/data-providers/dune/provider.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    console.log(
+      `\n dune_execution_id is ${execution_id} initial state is ${state} \n depending on size of query and Dune response time, this can take max 30 mins \n`
+    );
+
+    await this.getExecutionStatus(execution_id);
+    const results = await this.getResults(execution_id);
+    return this.formatData(results.result.rows, duneEthAddressColumn);
+  }
+
+  public async executeQueryCount(
```
Comment (MartinGbz) [inline group-generators/helpers/data-providers/dune/provider.ts:43]: Same problem here, you need to create a type that contains all your arguments.

**Review by MartinGbz — COMMENTED** [review]
Hey @sampolgar I added some comments :)

File: group-generators/helpers/data-providers/dune/provider.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+//Dune requires an API key
+export class DuneProvider {
+  /**
+   * Use this method to query Dune Analytics for your data.
+   * @param queryId The Dune QueryID e.g. 2034748 https://dune.com/queries/2034748"
+   * @param duneEthAddressColumn e.g. “The ETH Address Winner”  for the Dune Query https://dune.com/queries/2034748 -> used to match the ETH addresses
+   * @param queryParams optional! e.g. {address: '0x3CAaE25Ee616f2C8E13C74dA0813402eae3F496b', blocknumber: '66257668', chain: 'arbitrum'} for the Dune Query https://dune.com/queries/1618116
+   * @returns The FetchedData object in the <FetchedData> Type
+   */
+  public async executeQuery(
```
[the remark this answers is not in the document]
Comment (sampolgar) [reply group-generators/helpers/data-providers/dune/provider.ts]: Thanks @MartinGbz I've updated the args

File: group-generators/helpers/data-providers/dune/provider.ts (hunk 1 above)
Comment (sampolgar) [reply group-generators/helpers/data-providers/dune/provider.ts]: Thanks @MartinGbz, I've updated the args

File: group-generators/helpers/data-providers/dune/provider.ts [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    queryId: number,
+    duneEthAddressColumn: string,
```
[the remark this answers is not in the document]
Comment (sampolgar) [reply group-generators/helpers/data-providers/dune/provider.ts]: I understand. I'll add it in later

**Pull request merged**